R/print_anova.R

```diff
@@ -79,63 +79,89 @@ print_anova <- function(
   for(i in es) {x[[i]] <- printnum(x[[i]], digits = 3, gt1 = FALSE)}
   if(mse) x$mse <- printnum(x$mse, digits = 2)
 
-  # Assemble table
-  anova_table <- data.frame(x[, c("term", "statistic","df", "df_res", if(mse) "mse" else NULL, "p.value", es)], row.names = NULL)
+  # Assemble table -------------------------------------------------------------
+  cols <- intersect(
+    c("term", "statistic","df", "df_res", "mse", "p.value", es)
+    , colnames(x)
+  )
+  anova_table <- data.frame(x[, cols], row.names = NULL)
   anova_table[["term"]] <- prettify_terms(anova_table[["term"]])
 
-  ## Define appropriate column names
-  es_long <- c()
-  if("pes" %in% es) {
-    es_long <- c(es_long, "$\\hat{\\eta}^2_p$")
-  }
-  if("ges" %in% es) {
-    es_long <- c(es_long, "$\\hat{\\eta}^2_G$")
-  }
-  if("es" %in% es) {
-    es_long <- c(es_long, "$\\hat{\\eta}^2$")
-  }
-
-
-  mse_long <- if(mse) "$\\mathit{MSE}$" else NULL
   correction_type <- attr(x, "correction")
 
-  colnames(anova_table) <- c("Effect", "F", "df1", "df2", if(mse){"MSE"}else{NULL}, "p", es)
+  statistic <- attr(x, "statistic")
+  if(is.null(statistic)) statistic <- "F"
+  if(is.null(anova_table$df_res)) statistic <- "chisq"
+
+  names(es) <- es
+  renamers <- c(
+    term = "Effect"
+    , statistic = statistic   # defaults to "F" for backwards compatibility
+    , df = if(is.null(anova_table$df_res)){ "df" } else { "df1" }
+    , df_res = "df2"
+    , mse = "MSE"
+    , p.value = "p"
+    , es
+  )
+
+  colnames(anova_table) <- renamers[colnames(anova_table)]
   class(anova_table) <- "data.frame"
 
   if(!is.null(correction_type) && correction_type != "none") {
     variable_label(anova_table) <- c(
-      "Effect" = "Effect"
-      , "F" = "$F$"
-      , "df1" = paste0("$\\mathit{df}_1^{", correction_type, "}$")
-      , "df2" = paste0("$\\mathit{df}_2^{", correction_type, "}$")
-      , "p" = "$p$"
-    )
+      "Effect"  = "Effect"
```

Comment:
Similar to the above. Do you think we can turn these into generally applicable lookup tables that we place in their own utility functions?

Reply:
I think for this lookup table it won't work, because we are going from the generic term to a method-specific term. But I think the process of renaming columns (last comment) and adding variable labels (this comment) might be done within one step, with the labels created from the original column name (or at least a best guess that may be altered within a specific `apa_print()` method). This way, we should be able to make this step rock solid.

Reply:
I see. If you can get it to work that way that would be great. Otherwise we can just keep it like this for now.

Reply:
I think we should keep it like that, right now, and I will try to file a new PR with a possible implementation, soon.